contracts/src/StakeTable.sol

```diff
@@ -410,24 +424,105 @@ contract StakeTable is AbstractStakeTable {
             revert InsufficientStakeBalance(0);
         }
 
-        // Verify that the validator can sign for that blsVK
-        bytes memory message = abi.encode(msg.sender);
-        BLSSig.verifyBlsSig(message, blsSig, blsVK);
-
-        // Verify that the exit escrow period is over.
-        if (currentEpoch() < node.exitEpoch + exitEscrowPeriod(node)) {
-            revert PrematureWithdrawal();
-        }
+        // // Verify that the exit escrow period is over.
+        // if (currentEpoch() < node.exitEpoch + exitEscrowPeriod(node)) {
+        //     revert PrematureWithdrawal();
+        // }
+        totalStake -= balance;
 
         // Delete the node from the stake table.
-        delete nodes[key];
+        delete nodes[msg.sender];
```

Comment:
Not something to change in this PR, but I think it is not right to unregister a user after a withdrawal. I would prefer to add a function for depositing more funds for an already existing user.  Let us discuss, especially in the context of stake delegation.

Reply:
i agree with you